system-addon/lib/HighlightsFeed.jsm

```diff
@@ -43,18 +44,40 @@ this.HighlightsFeed = class HighlightsFeed {
   }
 
   async fetchHighlights(broadcast = false) {
-    this.highlights = await NewTabUtils.activityStreamLinks.getHighlights();
-    for (let highlight of this.highlights) {
-      highlight.hostname = shortURL(Object.assign({}, highlight, {url: highlight.url}));
-      highlight.image = highlight.preview_image_url;
-      if (highlight.bookmarkGuid) {
-        highlight.type = "bookmark";
-      }
-    }
+    // Request more than the expected length to allow for items being removed by
+    // deduping against Top Sites or multiple history from the same domain, etc.
+    const numItems = HIGHLIGHTS_MAX_LENGTH * 5 + TOP_SITES_SHOWMORE_LENGTH;
```

Comment:
no need to do this math on every fetch #nano-optimization

Reply:
Moved to top as `MANY_EXTRA_LENGTH`